hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/compact/DeserializedGenericRecord.java

```diff
@@ -131,92 +131,187 @@ public Set<String> getFieldNames() {
         return objects.keySet();
     }
 
+    @Override
+    public boolean getBoolean(@Nonnull FieldDescriptor fd) {
+        return getNonNull(fd.getFieldName(), "Boolean");
+    }
+
```

Comment:
Why do we need these methods on something we return to the user? 

```
 * ---- {@link DeserializedGenericRecord}         -> concrete class passed to the user
```

Reply:
🤔 I don't know exactly, SQL nested support seems to use it internall. So, it means it is returned to the user as well as used internally. 

You can debug a nested field SQL test and a `DeserializedGenericRecord` end up in the `FieldOperations` class.

Reply:
If it is the case, we should definitely fix it. This class is not meant to be used in SQL or queries, because it performs full deserialization, and we definitely want partial deserialization in queries

Reply:
The fix PR is merged as you know.